### DISCUSSION TITLE
PasswordHasher issue

### DISCUSSION THREAD

**Discussion by cybernet** [symfony/symfony#51047] [Q&A]
### Symfony version(s) affected

6.3.1

### Description

Using maker bundle ( 1.50 ) I created a registration form and controller, when I access it
I get the following error

```sh
Uncaught PHP Exception TypeError: 

"Symfony\Component\Form\Extension\PasswordHasher\Type\FormTypePasswordHasherExtension::__construct(): Argument #1 ($passwordHasherListener) must be of type 

Symfony\Component\Form\Extension\PasswordHasher\EventListener\PasswordHasherListener, 

Symfony\Component\Form\Extension\PasswordHasher\EventListener\PasswordHasherListener given, 

called in /home/dev/www/blb/var/cache/dev/ContainerRCJCv6k/getForm_TypeExtension_Form_PasswordHasherService.php on line 24" at /home/dev/www/blb/vendor/symfony/form/Extension/PasswordHasher/Type/FormTypePasswordHasherExtension.php line 25
{
    "exception": {}
}
```

I don't even know where to start to try and fix this ...

### How to reproduce

software info

PHP 8.2.8 (cli) (built: Jul  4 2023 14:52:54) (NTS gcc x86_64)

running symfony 6.3.1 with twig 3.6.1

### Additional Context

<img width="1424" alt="Screenshot 2023-07-20 at 05 31 38" src="https://github.com/symfony/symfony/assets/125879/759c4489-e70b-4713-9fa2-3d5681f1fb01">

**Comment by cybernet** [comment]
worth mentioning is that this appeared after I enabled `boostratp_5_layout` in twig

```yml
twig:
    default_path: '%kernel.project_dir%/templates'
    form_themes: ['bootstrap_5_layout.html.twig']
```

**Comment by xabbuh** [comment]
The error message doesn't make any sense to me as the expected class and the given class are the same. Do you have Opcache enabled? If so, can you clear it and check if the issue persists? Does clearing the application cache help?

**Reply by cybernet** [reply]
to my knowledge ( I may be wrong ) restarting php-fpm also clears OPcache cache, I ran cache:clear multiple times but still persists.

the problem "seems" to go away if I don't use `form_themes`. Also I've noticed something else. but I will need to post a video because I can't even find an appropriate {issue,discussion} title

**Comment by wouterj** [comment]
This sounds a lot like https://github.com/php/php-src/issues/9675#issuecomment-1342558074 and https://github.com/symfony/symfony/discussions/50152

This does not sound like a problem with the Symfony code, so I'm transforming this into a GitHub discussion.

**Comment by cybernet** [comment]
so far, disabling `OPcache` solves the issue - but the performance is appalling in `dev` `env` - instead of 20ms page load it goes up to 240ms
